projects/graph/src/graph.py

```diff
@@ -2,8 +2,40 @@
 Simple graph implementation compatible with BokehGraph class.
 """
 
-
+import random
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
-        pass  # TODO
+        self.vertices={}
+    def add_vertex(self,vertex_id):
+        self.vertices[vertex_id]=Vertex(vertex_id)
+    def add_edge(self,v1,v2):
+        if v1 in self.vertices and v2 in self.vertices:
+            self.vertices[v1].edges.add(v2)
+            self.vertices[v2].edges.add(v1)
+        else:
+            raise IndexError('That vertex does not exist.')
+    def bfs(self,starting_node):
```

Comment:
spot on. if you don't want to write a dfs method. just pass in a pop index to this method. `def bdft(self, starting_node, iPop=0):` then when you call the method you have the option of dfs (-1) or bfs(0).